### PR TITLE
*: Move bootstrap to CreateSession

### PR DESCRIPTION
**Pull request by shenli**
Bootstrap should be done for both tidb-server and embedded mode.

### CODE REVIEW

File: bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	}
+	_, err = s.Execute(fmt.Sprintf("CREATE DATABASE %s;", mysql.SystemDB))
+	if err != nil {
+		log.Fatal(err)
+	}
+	initUserTable(s)
+}
+
+func initUserTable(s *session) {
+	_, err := s.Execute(mysql.CreateUserTable)
```
Comment (siddontang) [inline bootstrap.go]: CreateUserTable can be moved here, not in mysqdef.

File: session.go [hunk 2]
```diff
@@ -398,5 +398,11 @@ func CreateSession(store kv.Storage) (Session, error) {
 
 	variable.BindSessionVars(s)
 	variable.GetSessionVars(s).SetStatusFlag(mysql.ServerStatusAutocommit, true)
+	b, ok := storeBootstrapped[store.UUID()]
```
Comment (siddontang) [inline session.go]: store.UUID may be not consistent, so if we restart tidb, we may re-run boostrap again.

Comment (shenli) [inline session.go]: I will check if system db exists in bootstrap(). If system db already exists, I will do nothing.

Comment (siddontang) [inline session.go]: got it

**Comment by siddontang** [comment]
I think we can support global variable table later.

**Comment by shenli** [comment]
Yes you can put variable into global stoage in bootstrap().

File: bootstrap.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	//  Check if system db exists.
+	_, err = s.Execute(fmt.Sprintf("USE %s;", mysql.SystemDB))
+	if err == nil {
+		// We have already finished bootstrap.
+		return
+	} else if !errors2.ErrorEqual(err, errors.ErrDatabaseNotExist) {
+		log.Fatal(err)
+	}
+	_, err = s.Execute(fmt.Sprintf("CREATE DATABASE %s;", mysql.SystemDB))
```
Comment (siddontang) [inline bootstrap.go]: you can warp a mustExecute function to simplify if err then log.Fatal.

Comment (shenli) [inline bootstrap.go]: Yes
PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**